src/python/pants/build_graph/target_filter_subsystem.py

```diff
@@ -0,0 +1,45 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+from builtins import object, set
+
+from pants.subsystem.subsystem import Subsystem
+
+
+logger = logging.getLogger(__name__)
+
+
+class TargetFilter(Subsystem):
+  """Filter targets matching configured criteria.
+
+  :API: public
+  """
+
+  options_scope = 'target-filter'
+
+  @classmethod
+  def register_options(cls, register):
+    super(TargetFilter, cls).register_options(register)
+
+    register('--exclude-tags', type=list,
+             default=[], fingerprint=True,
+             help='Skip targets with given tag(s).')
+
+  def apply(self, targets):
+    exclude_tags = set(self.get_options().exclude_tags)
+    return TargetFiltering(targets, exclude_tags).apply_tag_blacklist()
+
+
+class TargetFiltering(object):
+  """Apply filtering logic against targets."""
+
+  def __init__(self, targets, exclude_tags):
+    self.targets = targets
```

Comment:
I would have left targets as the input to `apply_tag_blacklist` since the subsystem (and this helper) are about the criteria. Configured with this criteria - once, filter these targets, now these and finally those. That said I'll merge and if you want to circle back - great, but also no need - this is style preference land.

Reply:
👍 (cross-linking for reference - this is done in https://github.com/pantsbuild/pants/pull/7280)